multi-node-test.sh

```diff
@@ -148,114 +107,106 @@ df -h" 2>&1 | dshbak -c; then
-        \"${nodes[2]}\",
-        \"${nodes[3]}\"
-    ],
-    \"host_list\": [\"${nodes[0]}\",
-        \"${nodes[1]}\",
-        \"${nodes[2]}\"
-    ],
-    \"use_daemon\": \"DvmRunner\",
-    \"log_base_path\": \"$log_base_path\"
-}
+
+mkdir -p ~/.config/avocado/sysinfo/
+cat <<EOF > ~/.config/avocado/sysinfo/commands
+ps axf
+dmesg
+df -h
 EOF
-    rm -rf $log_base_path/
-    python3 test_runner config=scripts/cart_multi_three_node.cfg \\
-        ${JENKINS_TEST_LIST_3[*]} || {
-        rc=\${PIPESTATUS[0]}
-        echo \"Test exited with \$rc\"
-    }
-    find $log_base_path/testRun -name subtest_results.yml \\
-         -exec grep -Hi fail {} \\;
-elif [ \"$1\" = \"5\" ]; then
-    cat <<EOF > scripts/cart_multi_five_node.cfg
-{
-    \"with_test_runner_host_list\": [
-        \"${nodes[1]}\",
-        \"${nodes[2]}\",
-        \"${nodes[3]}\",
-        \"${nodes[4]}\",
-        \"${nodes[5]}\"
-    ],
-    \"host_list\": [\"${nodes[0]}\",
-        \"${nodes[1]}\",
-        \"${nodes[2]}\",
-        \"${nodes[3]}\",
-        \"${nodes[4]}\"
-    ],
-    \"use_daemon\": \"DvmRunner\",
-    \"log_base_path\": \"$log_base_path\"
-}
+
+# apply fix for https://github.com/avocado-framework/avocado/issues/2908
+sudo ed <<EOF /usr/lib/python2.7/site-packages/avocado/core/runner.py
+/TIMEOUT_TEST_INTERRUPTED/s/[0-9]*$/60/
+wq
+EOF
+# apply fix for https://github.com/avocado-framework/avocado/pull/2922
+if grep \"testsuite.setAttribute('name', 'avocado')\" \
+    /usr/lib/python2.7/site-packages/avocado/plugins/xunit.py; then
+    sudo ed <<EOF /usr/lib/python2.7/site-packages/avocado/plugins/xunit.py
+/testsuite.setAttribute('name', 'avocado')/s/'avocado'/os.path.basename(os.path.dirname(result.logfile))/
+wq
 EOF
-    rm -rf $log_base_path/
-    python3 test_runner config=scripts/cart_multi_five_node.cfg \\
-        ${JENKINS_TEST_LIST_5[*]} || {
-        rc=\${PIPESTATUS[0]}
-        echo \"Test exited with \$rc\"
-    }
-    find $log_base_path/testRun -name subtest_results.yml \\
-         -exec grep -Hi fail {} \\;
 fi
+
+# put yaml files back
+restore_dist_files() {
+    local dist_files=\"\$*\"
+
+    for file in \$dist_files; do
+        if [ -f \"\$file\".dist ]; then
+            mv -f \"\$file\".dist \"\$file\"
+        fi
+    done
+
+}
+
+# set our machine names
+mapfile -t yaml_files < <(find \"$TESTDIR\" -name \\*.yaml)
+
+trap 'set +e; restore_dist_files \"\${yaml_files[@]}\"' EXIT
+
+# shellcheck disable=SC2086
+sed -i.dist -e \"s/- boro-A/- ${nodes[0]}/g\" \
+            -e \"s/- boro-B/- ${nodes[1]}/g\" \
+            -e \"s/- boro-C/- ${nodes[2]}/g\" \
+            -e \"s/- boro-D/- ${nodes[3]}/g\" \
+            -e \"s/- boro-E/- ${nodes[4]}/g\" \
+            -e \"s/- boro-F/- ${nodes[5]}/g\" \
+            -e \"s/- boro-G/- ${nodes[6]}/g\" \
+            -e \"s/- boro-H/- ${nodes[7]}/g\" \"\${yaml_files[@]}\"
+
+# let's output to a dir in the tree
+rm -rf \"$TESTDIR/avocado\" \"./*_results.xml\"
+mkdir -p "$LOGDIR"
```

Comment:
(lint) This is actually an end quote, but due to next char it looks suspect. [SC1079]
(lint) The surrounding quotes actually unquote this. Remove or escape them. [SC2027]
(lint) Double quote to prevent globbing and word splitting. [SC2086]